### ISSUE TITLE
Bootstrap upgrade results in "remote error: tls: handshake failure"

### ISSUE THREAD

**Issue by 2fst4u** [fluxcd/flux2#5122]
### Describe the bug

I was previously on fluxV2 version 0.37.0 and hadn't upgraded due to a longstanding issue that I have only just resolved. I now want to upgrade.

I've read through as many docs as possible and can't find specifically why this isn't working for me, but after upgrading the fluxcli and running bootstrap again I get the following error:

```
► connecting to github.com
► cloning branch "main" from Git repository "https://github.com/2fst4u/Kubernetes.git"
✔ cloned repository
► generating component manifests
✔ generated component manifests
✔ component manifests are up to date
✔ reconciled components
► determining if source secret "flux-system/flux-system" exists
► generating source secret
► applying source secret "flux-system/flux-system"
✔ reconciled source secret
► generating sync manifests
✔ generated sync manifests
✔ sync manifests are up to date
► applying sync manifests
✔ reconciled sync configuration
◎ waiting for GitRepository "flux-system/flux-system" to be reconciled
✗ failed to checkout and determine revision: unable to clone 'https://github.com/2fst4u/Kubernetes.git': Get "https://github.com/2fst4u/Kubernetes.git/info/refs?service=git-upload-pack": remote error: tls: handshake failure
◎ waiting for Kustomization "flux-system/flux-system" to be reconciled
✗ Source artifact not found, retrying in 30s
► confirming components are healthy
✔ helm-controller: deployment ready
✔ image-automation-controller: deployment ready
✔ image-reflector-controller: deployment ready
✔ kustomize-controller: deployment ready
✔ notification-controller: deployment ready
✔ source-controller: deployment ready
✔ all components are healthy
✗ bootstrap failed with 2 health check failure(s): [error while waiting for GitRepository to be ready: 'failed to checkout and determine revision: unable to clone 'https://github.com/2fst4u/Kubernetes.git': Get "https://github.com/2fst4u/Kubernetes.git/info/refs?service=git-upload-pack": remote error: tls: handshake failure', error while waiting for Kustomization to be ready: 'Source artifact not found, retrying in 30s']
```

I can see there's a redirect to info/refs?service=git-upload-pack which I don't understand why and can't find reference to.

### Steps to reproduce

1. Install v0.37.0
2. bootstrap to latest using the docs

### Expected behavior

bootstrap upgrades

### OS / Distro

WSL Ubuntu 22.04.5 LTS

### Flux version

flux: v2.4.0

### Flux check

► checking prerequisites
✔ Kubernetes 1.31.3+k3s1 >=1.28.0-0
► checking version in cluster
✔ distribution: flux-v2.4.0
✔ bootstrapped: true
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v1.1.0
✔ image-automation-controller: deployment ready
► ghcr.io/fluxcd/image-automation-controller:v0.39.0
✔ image-reflector-controller: deployment ready
► ghcr.io/fluxcd/image-reflector-controller:v0.33.0
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v1.4.0
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v1.4.0
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v1.4.1
► checking crds
✔ alerts.notification.toolkit.fluxcd.io/v1beta3
✔ buckets.source.toolkit.fluxcd.io/v1
✔ gitrepositories.source.toolkit.fluxcd.io/v1
✔ helmcharts.source.toolkit.fluxcd.io/v1
✔ helmreleases.helm.toolkit.fluxcd.io/v2
✔ helmrepositories.source.toolkit.fluxcd.io/v1
✔ imagepolicies.image.toolkit.fluxcd.io/v1beta2
✔ imagerepositories.image.toolkit.fluxcd.io/v1beta2
✔ imageupdateautomations.image.toolkit.fluxcd.io/v1beta2
✔ kustomizations.kustomize.toolkit.fluxcd.io/v1
✔ ocirepositories.source.toolkit.fluxcd.io/v1beta2
✔ providers.notification.toolkit.fluxcd.io/v1beta3
✔ receivers.notification.toolkit.fluxcd.io/v1
✔ all checks passed

### Git provider

github private repo

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by 2fst4u** [comment]
Also I have deleted the secret with `kubectl -n flux-system delete secret flux-system` and no change. My bootstrap command is `flux bootstrap github   --token-auth   --owner=2fst4u   --repository=Kubernetes   --branch=main   --path=./clusters/production   --personal   --components-extra=image-reflector-controller,image-automation-controller`

**Comment by 2fst4u** [comment]
I did an uninstall (with keep namespace just in case) and bootstrapped again and it's taking notably longer on "◎ waiting for GitRepository "flux-system/flux-system" to be reconciled" which it wasn't doing before. Before it was almost instant to fail out and it's currently taken about 2 minutes and not moved on from this point. No idea what to do at this point.

**Comment by dlidstrom** [comment]
Sometimes I find what’s wrong when I check flux logs or inspect the logs of one of the controllers of flux.

**Comment by 2fst4u** [comment]
The logs just say TLS handshake error over and over. 

Have I missed something in my bootstrap command?

**Comment by dlidstrom** [comment]
My command is this:

```bash
$ GITHUB_USER=dlidstrom flux bootstrap github \
  --read-write-key \
  --components-extra=image-reflector-controller,image-automation-controller \
  --owner=$GITHUB_USER \
  --repository=myrepo \
  --branch=main \
  --path=clusters \
  --personal
```

I think the token is defined in an environment variable.

**Comment by 2fst4u** [comment]
The only difference you have is `--read-write-key`. I added that and had no change.

I've done a complete uninstall (including namespace) and tried again and still no change. I'm at a loss as to what I'm missing here.

**Comment by 2fst4u** [comment]
When I manually access `https://github.com/2fst4u/Kubernetes.git/info/refs?service=git-upload-pack` using my username and PAT I also get an insecure TLS warning, including on my phone on data (so outside my network). Is there something wrong with my repo?

**Comment by 2fst4u** [comment]
`token-auth=false` gives the same error.

There has to be something I'm doing wrong here?

**Comment by 2fst4u** [comment]
I guess I can but should it be necessary?

**Comment by dlidstrom** [comment]
I guess not. I have upgraded once (from 2.2 to 2.4 I think it was). And it worked well. Good luck!

**Comment by 2fst4u** [comment]
Thank God I managed to find out what was wrong. One of my kubernetes nodes had a search domain set in the netplan config, which was messing up the DNS request. No idea why this is seemingly the only thing it was breaking but here we are.

This issue elsewhere lead me to it https://github.com/onedr0p/cluster-template/discussions/969

In my Ubuntu netplan config I needed to remove the search domain and it worked again.
